hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/hooks/HookEvmFrameState.java

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.service.contract.impl.state.hooks;
+
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HtsSystemContract.HTS_HOOKS_16D_CONTRACT_ADDRESS;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HtsSystemContract.HTS_HOOKS_16D_CONTRACT_ID;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.HookId;
+import com.hedera.node.app.service.contract.impl.exec.scope.HederaNativeOperations;
+import com.hedera.node.app.service.contract.impl.state.ContractStateStore;
+import com.hedera.node.app.service.contract.impl.state.DispatchingEvmFrameState;
+import com.hedera.node.app.service.contract.impl.state.WritableEvmHookStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.account.MutableAccount;
+import org.hyperledger.besu.evm.code.CodeFactory;
+
+/**
+ * EVM frame state used during hook execution. For address 0x16d, it returns
+ * the executing hook's contract bytecode (fetched from the hook store).
+ * TODO: Additional behavior like storage redirection and debit-from-owner will be added here in next PRs
+ */
+public class HookEvmFrameState extends DispatchingEvmFrameState {
+    private final WritableEvmHookStore evmHookStore;
+    private final HookId hookId;
+    private final CodeFactory codeFactory;
+
+    /**
+     * @param nativeOperations the Hedera native operation
+     * @param contractStateStore the contract store that manages the key/value states
+     */
+    public HookEvmFrameState(
+            @NonNull final HederaNativeOperations nativeOperations,
+            @NonNull final ContractStateStore contractStateStore,
+            @NonNull final CodeFactory codeFactory,
+            @NonNull final WritableEvmHookStore evmHookStore,
```

Comment:
As noted elsewhere, I think one of the other classes passed in here will likely be responsible for using the `WritableEvmHookStore` as necessary.

Reply:
Removed and injected Hook